shell/probe/nocturnal.sh

```diff
@@ -1,30 +1,32 @@
 #!/bin/bash
 
-PRELUDE_API=${PRELUDE_API:="https://api.preludesecurity.com"}
+PRELUDE_SLEEP=${PRELUDE_SLEEP:=14440}
+PRELUDE_DIR=${PRELUDE_DIR:="./cache"}
+PRELUDE_CA=${PRELUDE_CA:="prelude-account-prod-us-west-1.s3.amazonaws.com"}
 
-sys=$(uname -s)-$(uname -m)
-dos=$(echo $sys | tr '[:upper:]' '[:lower:]')
+api="https://api.preludesecurity.com"
+dos=$(echo $(uname -s)-$(uname -m))
 
 while :
-do
-    temp=$(mktemp)
-    location=$(curl -sL -w %{url_effective} -o $temp -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $PRELUDE_API)
-    test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)
+do  
+    exe=$PRELUDE_DIR/$(uuidgen)
+    location=$(curl -sL -w %{url_effective} --create-dirs -o $exe -H "token:${PRELUDE_TOKEN}" -H "dos:${dos}" -H "dat:${dat}" $api)
+    test=$(echo $location | grep -o '[0-9a-f]\{8\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{4\}-[0-9a-f]\{12\}' | head -n 1)    
+    unset dat
```

Comment:
this unset should be near the sleep, no?

Reply:
could go either way; not sure I care but will examine both

Reply:
thing I'm thinking about that needs the unset near the sleep:

iteration 1:
 - get a test, set dat={123-123-123:100}
 iteration 2:
 - send in that dat and get back "no more tests"
 - go to sleep
 iteration 3:
- exit sleep, dat still set to {123-123-123:100}
- send in that dat while getting next round of tests

Reply:
on iteration 2, when you get 'no more tests', you'd be unsetting the 'data' before getting to the if/check on the $test.

Reply:
(but I swapped to sleep anyway, as I think it reads better)